src/Microsoft.AspNetCore.Sockets.Client.Http/ServerSentEventsTransport.cs

```diff
@@ -83,67 +81,67 @@ public Task StartAsync(Uri url, Channel<byte[], SendMessage> application, Transf
-                    {
-                        var parseResult = _parser.ParseMessage(input, out consumed, out examined, out var buffer);
-
-                        switch (parseResult)
+                        try
                         {
-                            case ServerSentEventsMessageParser.ParseResult.Completed:
-                                _application.Writer.TryWrite(buffer);
-                                _parser.Reset();
-                                break;
-                            case ServerSentEventsMessageParser.ParseResult.Incomplete:
-                                if (result.IsCompleted)
-                                {
-                                    throw new FormatException("Incomplete message.");
-                                }
-                                break;
+                            var parseResult = _parser.ParseMessage(input, out consumed, out examined, out var buffer);
+
+                            switch (parseResult)
+                            {
+                                case ServerSentEventsMessageParser.ParseResult.Completed:
+                                    await _application.Output.WriteAsync(buffer);
+                                    _parser.Reset();
+                                    break;
+                                case ServerSentEventsMessageParser.ParseResult.Incomplete:
+                                    if (result.IsCompleted)
+                                    {
+                                        throw new FormatException("Incomplete message.");
+                                    }
+                                    break;
+                            }
+                        }
+                        finally
+                        {
+                            pipelineReader.AdvanceTo(consumed, examined);
                         }
-                    }
-                    finally
-                    {
-                        pipelineReader.AdvanceTo(consumed, examined);
                     }
                 }
-            }
-            catch (OperationCanceledException)
-            {
-                _logger.ReceiveCanceled();
-            }
-            finally
-            {
-                readCancellationRegistration.Dispose();
-                _transportCts.Cancel();
-                stream.Dispose();
-                _logger.ReceiveStopped();
+                catch (OperationCanceledException)
+                {
+                    _logger.ReceiveCanceled();
+                }
+                finally
+                {
+                    readCancellationRegistration.Dispose();
+                    _transportCts.Cancel();
+                    _logger.ReceiveStopped();
+                }
             }
         }
 
         public async Task StopAsync()
         {
             _logger.TransportStopping();
             _transportCts.Cancel();
-            _application.Writer.TryComplete();
```

Comment:
Why is this gone?

Reply:
It isn't required. All completion happens in here now https://github.com/aspnet/SignalR/pull/1435/files/c34463436737ee897219649a4e61c56e8f646346#diff-579fedec848db9fecbd95f172d7654e7R66. Both tasks should always end and we should clean up in the same place.